src/services/DBusService.cpp

```diff
@@ -0,0 +1,105 @@
+#include "services/DBusService.hpp"
+
+namespace waybar::services {
+
+DBusService::DBusService(const Json::Value& config, const std::string& name, const std::string& id,
+                         const std::string& format)
+    : Config(config, format),
+      name_(std::move(name)),
+      introspection_data_{Gio::DBus::NodeInfo::create_for_xml(
+          Glib::ustring::sprintf(DBUS_INTROSPECTION_XML, name_))} {
+  owner_id_ = Gio::DBus::own_name(
+      Gio::DBus::BusType::BUS_TYPE_SESSION, Glib::ustring::sprintf(DBUS_NAME, name_),
```

Comment:
Service name does not include `id` or bar id. What does it do when you have multiple instances of the same module? I.e. `clock#first`, `clock#second`.

What would it do when you have two bar configs with differently configured clock modules? I.e.
```json
[
  {
    "position": "top",
    "modules-center": ["clock"],
    "clock": {
        "timezone": "America/Los_Angeles"
    }
  },
  {
    "position": "bottom",
    "modules-center": ["clock"],
    "clock": {
        "timezone": "UTC"
    }
  }
]
```

Reply:
Agree. Totally forgot about this . Implementation took huge time.